codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -20,83 +20,128 @@
 import io.netty.handler.codec.compression.ZlibWrapper;
 import io.netty.util.internal.StringUtil;
 
+import java.util.regex.Pattern;
+
 /**
  * Compresses an {@link HttpMessage} and an {@link HttpContent} in {@code gzip} or
- * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header.
- * If there is no matching encoding, no compression is done.  For more
- * information on how this handler modifies the message, please refer to
- * {@link HttpContentEncoder}.
+ * {@code deflate} encoding while respecting the {@code "Accept-Encoding"} header. If there is no matching encoding,
+ * no compression is done.
+ *
+ * Note that only a {@link FullHttpResponse} or a response with {@code "Transfer-Encoding: chunked"}
+ * will be compressed. Also the {@code "Content-Length"} and {@code "Content-Type"} are checked against
+ * the given <tt>minCompressableContentLength</tt> and <tt>compressableContentTypes</tt> to determined if
+ * a compression would be effective at all.
+ *
+ * For more information on how this handler modifies the message, please refer to {@link HttpContentEncoder}.
  */
 public class HttpContentCompressor extends HttpContentEncoder {
 
+    public static final int DEFAULT_COMPRESSION_LEVEL = 6;
+    public static final int DEFAULT_WINDOW_BITS = 15;
+    public static final int DEFAULT_MEM_LEVEL = 8;
+    public static final int DEFAULT_MIN_COMPRESSABLE_CONTENT_LENGTH = 1024;
+    public static final Pattern DEFAULT_COMPRESSABLE_CONTENT_TYPES = Pattern.compile("(text/.*|application/json.*)");
```

Comment:
To maximize backward compatibility, we could rename DEFAULT_COMPRESSABLE_CONTENT_TYPES to "RECOMMENDED_COMPRESSABLE_CONTENT_TYPES" and use null as default value. I could then change beginEncode to always compress in this case (independet of the presence of a FullResponse or chunked transfer) - However, this would also still create invalid responses when used incorrectly (which is quite easy :-/ )